Provision Tracks with SwiftPM instead of CocoaPods

mokagio opened the pull request:
## Description
What it says on the title. See also:

- https://linear.app/a8c/issue/AINFRA-398 which tracks this task
- https://linear.app/a8c/issue/AINFRA-216 which tracks the whole CocoaPods removal effort
- https://linear.app/a8c/issue/AINFRA-471 which tracks moving Experiments from a project to a Swift package, enabled by this change

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
N.A. 

## Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
See green prototype build, or checkout the branch locally and verify the app builds.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
N.A.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. — N.A.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

mokagio committed:
Provision Tracks from CocoaPods to SwiftPM

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/244">22.4</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio commented on Experiments/Experiments.xcodeproj/project.pbxproj:
```diff
@@ -13,6 +13,7 @@
 		0270C0A327069B7800FC799F /* FeatureFlagService.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0270C0A227069B7800FC799F /* FeatureFlagService.swift */; };
 		0270C0A527069B8900FC799F /* DefaultFeatureFlagService.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0270C0A427069B8900FC799F /* DefaultFeatureFlagService.swift */; };
 		0270C0A727069BA500FC799F /* BuildConfiguration.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0270C0A627069BA500FC799F /* BuildConfiguration.swift */; };
+		3F8115032DD2A6EC0042DDB6 /* AutomatticTracks in Frameworks */ = {isa = PBXBuildFile; productRef = 3F8115022DD2A6EC0042DDB6 /* AutomatticTracks */; };
```
Notice the project file and the shared scheme will disappear as soon as Experiments will become a package.

mokagio commented on WooCommerce.xcworkspace/xcshareddata/swiftpm/Package.resolved:
```diff
@@ -73,15 +55,6 @@
-          "revision": "7795df4fff1a9cd231fe4867ae54f4dc5f5734f9",
-          "version": "13.7.1"
-        }
-      },
```
I'm not sure why these dependencies disappeared from the workspace resolved file in this commit. I would have expect that to happen as part of #15612 🤔

mokagio committed:
Track test plan for Experiments, used to run the tests just for it

mokagio committed:
Add Tracks to the Woo and Experiments test targets

Otherwise, they won't compile!

See
https://buildkite.com/automattic/woocommerce-ios/builds/29791/steps?jid=0196c741-beca-4d2d-a36b-9b968d8e99c3

mokagio commented on Experiments/Experiments.xcodeproj/project.pbxproj:
```diff
@@ -270,10 +281,14 @@
 				"${PODS_ROOT}/Target Support Files/Pods-ExperimentsTests/Pods-ExperimentsTests-frameworks-${CONFIGURATION}-input-files.xcfilelist",
 			);
+			inputPaths = (
+			);
```
With the CocoaPods removal, we'll also stop seeing these sections being removed, then added back, then removed...
